tags/TagMap.md

```diff
@@ -23,23 +23,43 @@ and group stats, annotate traces and logs.
 `TagValue` is a string. It MUST contain only printable ASCII (codes between
 32 and 126)
 
-## TagScope
+## TagMetadata
 
-`TagScope` is used to determine the scope of a `Tag`. The values for the `TagScope` are
-Local or Request. In future, additional values can be added to address specific situations.
+`TagMetadata` contains properties associated with a `Tag`. For now the only property `TagTTL`
+is defined. In future, additional properties may be added to address specific situations.
 
-The tag creator determines the scope of the tag.
+A tag creator determines metadata of a tag it creates.
 
-**Local Scope**
-Tag with `Local` scope are used within the process it created. Such tags are not propagated
-across process boundaries. Even if the process is reentrant the tag MUST be excluded from
-propagation when the call leaves the process.
+### TagTTL
 
-**Request Scope**
-If a tag is created with the `Request` scope then it is propagated across process boundaries subject 
-to outgoing and incoming (on remote side) filter criteria. See `TagPropagationFilter` in 
-[Tag Propagation](#Tag Propagation). Typically 'Request' tags represents a request, processing
- of which may span multiple entities.
+`TagTTL` is an integer that represents number of hops a tag can propagate. Anytime a sender serializes a tag
+, sends it over the wire and receiver unserializes the tag then the tag is considered to have travelled one hop. 
+There could be one or more proxy(ies) between sender and receiver. Proxies are treated as transparent
+entity and they do not create additional hops.
+
+Upon receiving tag from remote entity a tag extractor
+
+- MUST decrement the value of `TagTTL` by one if it is greater than zero.
+- MUST discard the `Tag` if the `TagTTL` value is zero.
+- MUST not change the value of `TagTLL` if it is -1.
+
+Upon sending tag to remote entity a tag extractor
+- MUST send `TagTTL` ONLY if its value is greater than 0.
+
+Absence of TagTTL on the wire is treated as having TagTTL of -1. This is for backward compatibility.
+
+For now, valid values of `TagTTL` are
+- **NO_PROPAGATION(0)**: Tag with `TagTTL` value of zero is considered to have local scope and
+ is used within the process it created. 
+- **UNLIMITED_PROPAGATION(-1)**: Tag with `TagTTL` value of -1 can propagate unlimited hops.
+ However, it is still subject to outgoing and incoming (on remote side) filter criteria. 
+ See `TagPropagationFilter` in [Tag Propagation](#Tag Propagation). Tag with `TagTTL` value of -1
+ is used to represent a request, processing of which may span multiple entities.
+
+## Tag Conflict Resolution
+If a new tag conflicts with an existing tag then the new tag takes precedence. Entire `Tag` along 
+with `TagValue` and `TagMetadata` is overwritten with a most recent tag (either it is locally
```

Comment:
I think 'shadowed by' or similarly is better than 'overwritten with', which implies an update and obscures the fact that the original tag will be visible again once the new tag goes out of scope.

Reply:
made it more clearer with an example.